tests/test_domains_config.py

```diff
@@ -168,6 +168,27 @@ def test_get_volume_config_error(domains_config):
     assert namespace == None
 
 
+def test_volume_config_hostname_worker_id_cache(domains_config):
+    dc = domains_config
+    worker_id = 'worker1'
+    hostname = 'example007.com'
+    dc._cwm_api_volume_configs['hostname:{}'.format(hostname)] = {
+        'instanceId': worker_id, 'minio_extra_configs': {'hostnames': [{'hostname': hostname}]}
+    }
+    volume_config = dc.get_cwm_api_volume_config(hostname=hostname)
+    assert volume_config.id == worker_id
+    assert domains_config.keys.volume_config_hostname_worker_id.get(hostname).decode() == worker_id
+
+    # empty volume configs and try to get worker_id from cache
+    volume_config = dc._cwm_api_volume_configs['hostname:{}'.format(hostname)] = {}
```

Comment:
all the code from this line and until the end of the function doesn't call the actual function, it just tests internal redis / dicts which are irelevant for this test

you need to set `dc._cwm_api_volume_configs['hostname:{}'.format(hostname)] = {}`
then make another call to get_cwm_api_volume_config which should return the relevant volume config from the cache

Reply:
unit tests needs to test the relevant functionality which your PR is implementing, striving for 100% coverage of all relevant code paths